source/maze/MazeIO.cpp

```diff
@@ -89,9 +89,9 @@ void MazeIO::loadMaze(char* fileName)
                 maze->startPos.col = j / 2;
                 wallToPut = wallError;
                 continue;
-            case 'G':
-                maze->goalPos.row = i / 2;
-                maze->goalPos.col = j / 2;
+            case 'G':		/* needs to be tested */
```

Comment:
This is *definitely* no-no! Don't try ~~lazy~~ workaround.
You must not assume there are 4 goal in a given maze file.
when reading a maze, you must empty `goalPos` and fill fill goals by using [`vector::push_back()`](http://www.cplusplus.com/reference/vector/vector/push_back/)